packages/atomic/src/components/search/result-template-components/quickview-preview-bar/quickview-preview-bar.ts

```diff
@@ -0,0 +1,57 @@
+import {QuickviewWordHighlight} from '../quickview-word-highlight/quickview-word-highlight';
+
+const buildPreviewBar = (documentWriter: Document) => {
+  const bar = documentWriter.createElement('div');
+  bar.style.position = 'fixed';
+  bar.style.top = '0';
+  bar.style.right = '0';
+  bar.style.width = '15px';
+  bar.style.height = '100%';
+  return bar;
+};
+
+const buildPreviewUnit = (
+  documentWriter: Document,
+  word: QuickviewWordHighlight,
+  wordElement: HTMLElement,
+  docHeight: number
+) => {
+  const elementPosition = wordElement.getBoundingClientRect().top;
+  const previewUnit = documentWriter.createElement('div');
+  previewUnit.style.position = 'absolute';
+  previewUnit.style.top = `${(elementPosition / docHeight) * 100}%`;
+  previewUnit.style.width = '100%';
+  previewUnit.style.height = '1px';
+  previewUnit.style.border = `1px solid ${word.saturateColor}`;
+  previewUnit.style.backgroundColor = word.color;
+  return previewUnit;
+};
+
+export const buildQuickviewPreviewBar = (
```

Comment:
IMO I would prefer a name like `buildQuickviewScrollbarHighlights` or `buildQuickviewHighlightedScrollbar`, but I don't feel strongly about it.

Reply:
Ok, I like the shorter name, and would be tempted to leave it like this.

But if other people have an opinion on it I will change it.